### PR TITLE
Added report comment and post API calls

### LINKED ISSUE #3: Add "Report" button for content
## Introduction
Add the ability to report various types of content to the app.

## Requirements
- Ability to report posts via context menu
- Ability to report comments via context menu

## Discussion
We need this before we try a pass at App Store submission.

## What I Need
Some gosh darn report buttons.

### PR DESCRIPTION
**Pull request by JakeShirley**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] This PR addresses one or more open issues that were assigned to me:
      - #3
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
This PR adds reporting for posts and comments via a similar popup sheet as posting a new post.

It also adds backing networking calls for those scenarios.

## Additional Context
n/a

---

fixes #3

### CODE REVIEW

**Review by WestonHanners — APPROVED** [review]
Can we test this on a live instance?

**Review by WestonHanners — APPROVED** [review]

**Comment by WestonHanners** [comment]
Ignore the action failures, I removed the analyze action till we can fix the package bugs

**Review by EricBAndrews — APPROVED** [review]

File: Mlem/API/Models/Posts/APIPostReport.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import Foundation
+
+// lemmy_db_schema::source::post::PostReport
+struct APIPostReport: Decodable {
+    let id: Int
+    let creatorId: Int
+    let postId: Int
+    let originalPostName: String
+    let originalPostUrl: String?
```
Comment (ShadowJonathan) [inline Mlem/API/Models/Posts/APIPostReport.swift:16]: This can be a `URL?`, but this is also alright

**Review by ShadowJonathan — APPROVED** [review]
API sections LGTM

File: Mlem/API/Models/Posts/APIPostReport.swift (hunk 1 above)
Comment (JakeShirley) [reply Mlem/API/Models/Posts/APIPostReport.swift:16]: Ah good catch.

**Review by EricBAndrews — APPROVED** [review]

**Pull request merged**